jbpm-document/src/main/java/org/jbpm/document/marshalling/DocumentMarshallingStrategy.java

```diff
@@ -48,40 +52,22 @@ public Document buildDocument( String name, long size, Date lastModified, Map<St
 
     @Override
     public void write(ObjectOutputStream os, Object object) throws IOException {
-        Document document = (Document) object;
-
-        if (document != null && document.getContent() != null) {
-            documentStorageService.saveDocument(document, document.getContent());
-        }
-        os.writeUTF(document.getIdentifier());
-        os.writeUTF(document.getClass().getCanonicalName());
-        os.writeUTF(document.getLink());
+        throw new UnsupportedOperationException("write is not supported anymore, use marshal instead");
     }
 
     public Object read(ObjectInputStream os) throws IOException, ClassNotFoundException {
-        String objectId = os.readUTF();
-        String canonicalName = os.readUTF();
-        String link = os.readUTF();
-        try {
-            Document doc = documentStorageService.getDocument(objectId);
-            Document document = (Document) Class.forName(canonicalName).newInstance();
-            document.setIdentifier(objectId);
-            document.setLink(link);
-            document.setName(doc.getName());
-            document.setSize(doc.getSize());
-            document.setLastModified(doc.getLastModified());
-            document.setAttributes(doc.getAttributes());
-            document.setContent(doc.getContent());
-            return document;
-        } catch(Exception e) {
-            throw new RuntimeException("Cannot read document", e);
-        }
+        throw new UnsupportedOperationException("read is not supported anymore, use unmarshal instead");
     }
 
     @Override
     public byte[] marshal(Context context, ObjectOutputStream objectOutputStream, Object o) throws IOException {
         Document document = (Document) o;
-        documentStorageService.saveDocument(document, document.getContent());
+        String updatedAttribute = document.getAttribute(Document.UPDATED_ATTRIBUTE);
+        if ("true".equals(updatedAttribute)) {
```

Comment:
better Boolean.parseBoolean(String s) cause it would handle null for you as well